crates/shared/src/price_estimation/trade_verifier.rs

```diff
@@ -400,16 +463,89 @@ fn encode_settlement(
         &query.in_amount.get(),
     );
 
-    EncodedSettlement {
+    let mut trades = vec![encoded_trade];
```

Comment:
At this point the function becomes pretty big so refactoring it into a few functions would be good.

Reply:
I just tried to refactor the code, and it produced many changes, which would overcomplicate the PR and, as a result, make it harder to grasp the real changes. I would suggest making the refactoring as a separate PR either  **before** this PR or when removing the legacy trade. Wdyt?

Reply:
Given that the price vector will always be very small (<10) I think it would be totally fine to do a simple linear search through the vector to find the index.

Reply:
Extracted some logic into separate functions. Should be now a bit more readable.